### ISSUE TITLE
Make bail behavior of tap-runner configurable

### ISSUE THREAD

**Issue by ericcornelissen** [stryker-mutator/stryker-js#4276]
**Is your feature request related to a problem? Please describe.**
The way [bailing is implemented the tap-runner](https://github.com/stryker-mutator/stryker-js/blob/ff87edfc4e9b611fba041017a8ed77415c58b503/packages/tap-runner/src/tap-helper.ts#L26-L31) is not compatible with all ways in which tests can be run. The top-level [disableBail](https://stryker-mutator.io/docs/stryker-js/configuration/#disablebail-boolean) option can be used to avoid such problems, but just disabling the tap-runner's "hard" bail functionality would suffice (and keep the main functionality of bail in place).

**Describe the solution you'd like**
Add a new option to the [tap-runner configuration](https://stryker-mutator.io/docs/stryker-js/tap-runner/#configuring) to be able to disable its "hard" bail functionality, e.g.:

```js
{
  "testRunner": "tap",
  "tap": {
    // etc...
    "hardBail": false
  }
}
```

**Describe alternatives you've considered**
As mentioned, the top-level [disableBail](https://stryker-mutator.io/docs/stryker-js/configuration/#disablebail-boolean) option can be used as a "workaround" but this is disables more than is absolutely necessary. No other alternatives were considered.

**Additional context**
See the discussion over at #3245 for more details and context.

**Comment by danny12321** [comment]
I've discussed this with @nicojs. I will implement this feature, but with the name `forceBail`, because we are forcing the process to be killed.

**Comment by nicojs** [comment]
You're _forcing_ me to make _hard_ choices, but I won't _bail_, I'll allow it 😉

**Comment by nicojs** [comment]
Released in 7.1. @ericcornelissen you should now be able to use ava 😲. Would you mind giving it a shot?

**Comment by ericcornelissen** [comment]
Works like a charm @nicojs :ok_hand:
